client_integration_test.go

```diff
@@ -1489,6 +1490,59 @@ func TestClientReadFrom(t *testing.T) {
 	}
 }
 
+// A wrongSizeReader reads zeros and has an arbitrary Size.
```

Comment:
`wrongSizeReader` does not actually read zeros, because it does not modify the byte slice that is passed into `Read` at all.

We should do one of the following:
* Change the documentation to say that it is a no-op that pretends to read `actual` number of bytes, and that one can have it report any arbitrary value for `Size()` by setting `reported`. (That is, describe the actual contract the implementation given fufills.)
* Actually zero out the byte-slice passed into `Read`, in accordance with the contract this comment purposes.
* Have this wrap an `io.Reader` and then we don’t have to specify what the `Read` does at all.

(nitpick) Hm, also, one of the `wrongSizeReaders` that we make is not a wrong size. 😅

Reply:
Replaced by your suggestion of wrapping an io.Reader. The code is shorter to boot.